documentation/Software Requirements Specification.md

```diff
@@ -0,0 +1,331 @@
+
+***Main scenario:***
+* User clicks the “Add” button in the cars list view mode; 
+* Application displays form to enter car data; 
+* User enters car data and presses “Save” button; 
+* If any data is entered incorrectly, incorrect data messages are displayed; 
+* If entered data is valid, then record is adding to database; 
+* If error occurs, then error message is displaying; 
+* If new car record is successfully added, then list of cars with added records is displaying.
+
+***Cancel operation scenario:***
+* User clicks the “Add” button in the cars list view mode; 
+* Application displays form to enter car data; 
+* User enters car data and presses “Cancel” button;
+* Data don’t save in data base, then list of cars records is displaying to user. 
+* If the user selects the menu item "Orders”, ”Clients” or "Cars", the data will not be saved to the database and the corresponding form with updated data will be opened.
+
+
+![](add_car.PNG)
+<br>Pic. 3.2 Add car
+
+When adding a car, the following details are entered:
+
+* Marka - car marka;
+* Model - car model;
+* Cost - the cost of the car;
+* Year of release - the year of the car 's release
+* Color - the color of the car
+
+Constraints for data validation:
+
+* Marka - maximum length of 45 character;
+* Model - maximum length of 45 character;
+* Cost - maximum length of 45 character;
+* Year of release - date of creation in the format dd/mm/yyyy.
+* Color - maximum length of 45 character;
+
+
+### 3.3 Edit car
+***Main scenario:***
+* User clicks the “Edit” button in the cars list view mode; 
+* Application displays form to enter car data; 
+* User enters car data and presses “Save” button; 
+* If any data is entered incorrectly, incorrect data messages are displayed; 
+* If entered data is valid, then edited data is added to database; 
+* If error occurs, then error message is displaying; 
+* If car record is successfully edited, then list of cars with added records is displaying. 
+
+Cancel operation scenario:
+
+* User clicks the “Edit” button in the cars list view mode; 
+* Application displays form to enter car data; 
+* User enters car data and presses “Cancel” button; 
+* Data don’t save in data base, then list of cars records is displaying to user.
+
+If the user selects the menu item "Orders”, ”Clients” or "Cars", the data will not be saved to the database and the corresponding form with updated data will be opened.
+
+![](edit_car.PNG)
+<br>Pic. 3.3 Edit car.
+
+
+### 3.4 Removing the car
+
+***Main scenario:***
+
+* The user, being in the list of machines mode, presses the "Delete" button in the selected row of machines;
+* The application displays a confirmation dialog box "Please confirm the removal of the machine ?”;”;
+* The user confirms the removal of the machine;
+* The record has been deleted from the database;
+* When an error occurs, an error message is displayed;
+* If the client record is successfully deleted, a list of machines without deleted records is displayed.
```

Comment:
The same as for `Client`deletion. How the deletion of the car will affect the `Orders` table?